paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -243,22 +243,72 @@ def current_value_forecast_policy(historical_load, **kwargs):
     return historical_load[-1][1]
 
 
+def window_historical_load(historical_load, window_begin, window_end):
+    """Filter historical_load down to just the datapoints lying between times window_begin and window_end, inclusive."""
+    filtered = []
+    for timestamp, value in historical_load:
+        if timestamp >= window_begin and timestamp <= window_end:
+            filtered.append((timestamp, value))
+    return filtered
+
+
+def trailing_window_historical_load(historical_load, window_size):
+    window_end, _ = historical_load[-1]
+    window_begin = window_end - window_size
+    return window_historical_load(historical_load, window_begin, window_end)
+
+
 @register_autoscaling_component('moving_average', FORECAST_POLICY_KEY)
 def moving_average_forecast_policy(historical_load, moving_average_window_seconds, **kwargs):
     """Does a simple average of all historical load data points within the moving average window. Weights all data
     points within the window equally."""
 
-    window_end, _ = historical_load[-1]
-    window_begin = window_end - moving_average_window_seconds
+    windowed_data = trailing_window_historical_load(historical_load, moving_average_window_seconds)
+    windowed_values = [value for timestamp, value in windowed_data]
+    return sum(windowed_values) / len(windowed_values)
```

Comment:
no; historical_load always has the most recent load measurement, which would fall within the window.